File: CS/Q148401/ChangeCaptionDialog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Columns;

namespace Q148401 {
    public partial class ChangeCaptionDialog : Form {
        private GridColumn column;

        public ChangeCaptionDialog() {
            InitializeComponent();
        }

        public ChangeCaptionDialog(GridColumn column, Point position) : this() {
            this.column = column;
            textBox1.Text = column.Caption;
            Location = position;
        }

        private void OnApplyClick(object sender, EventArgs e) {
            column.Caption = textBox1.Text;
            Close();
        }

        private void OnCancelClick(object sender, EventArgs e) {
            Close();
        }
    }
}

File: VB/Q148401/ChangeCaptionDialog.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Columns

Namespace Q148401
	Partial Public Class ChangeCaptionDialog
		Inherits Form
		Private column As GridColumn

		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal column As GridColumn, ByVal position As Point)
			Me.New()
			Me.column = column
			textBox1.Text = column.Caption
			Location = position
		End Sub

		Private Sub OnApplyClick(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			column.Caption = textBox1.Text
			Close()
		End Sub

		Private Sub OnCancelClick(ByVal sender As Object, ByVal e As EventArgs) Handles button2.Click
			Close()
		End Sub
	End Class
End Namespace